<?xml version="1.0" encoding="utf-8"?>
<launch>

   <node name="rviz" pkg="rviz" type="rviz" output="screen"
    args="-d $(find rl_control_test)/config/viz.rviz"> 
  </node>
  

  <!-- size of map, change the size in x, y, z according to your application -->
  <arg name="map_size_x" value="50.0"/>
  <arg name="map_size_y" value="50.0"/>
  <arg name="map_size_z" value="5.0"/>

  <!-- topic of your odometry such as VIO or LIO -->
  <arg name="odom_topic" value="/state_ukf/odom" />

  <!-- use simulator -->
    <include file="$(find rl_control_test)/launch/simulator.xml">
      <arg name="map_size_x_" value="$(arg map_size_x)"/>
      <arg name="map_size_y_" value="$(arg map_size_y)"/>
      <arg name="map_size_z_" value="$(arg map_size_z)"/>
      <arg name="c_num" value="0"/>
      <arg name="p_num" value="200"/>
      <arg name="odometry_topic" value="$(arg odom_topic)" />
    </include>


  <node pkg="rl_control_test" type="sfc_node" name="sfc_node" output="screen">
        <param name="reset_quad_topic" value="/quadrotor_simulator_so3/reset_quad_pos"/> 
        <param name="reset_map_topic" value="/random_forest/reset"/>    
        <param name="map_resetted_topic" value="/map_generator/map_updated"/>    
        <param name="map_topic" value="/map_generator/global_cloud"/>    
        <param name="target_topic" value="/move_base_simple/goal"/>
        <param name="odom_topic" value="$(arg odom_topic)"/>
        <param name="hpoly_srv" value="hpoly_srv"/>
        <param name="dilate_radius" value="0.5"/>
        <param name="voxel_width" value="0.25"/>
        <param name="rl_rounds" value="20"/>
        <param name="min_dist" value="20"/>
        <rosparam file="$(find rl_control_test)/config/sfc_node.yaml" command="load" />
  </node>

</launch>
